v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,233 @@
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  BASE_ENDPOINT
+} = require("./config");
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  defaultPutRequestConfig
+} = require("../../util");
+const { fetchKustomer, validateEvent } = require("./util");
+
+// Function responsible for constructing the Kustomer (User) Payload for identify
+// type of events.
+const constructKustomerPayload = (message, category) => {
+  const kustomerPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[category.name]
+  );
+
+  if (
+    !get(kustomerPayload, "name") &&
+    getFieldValueFromMessage(message, "firstName") &&
+    getFieldValueFromMessage(message, "lastName")
+  ) {
+    kustomerPayload.name = `${getFieldValueFromMessage(
+      message,
+      "firstName"
+    )} ${getFieldValueFromMessage(message, "lastName")}`;
+  }
+
+  if (getFieldValueFromMessage(message, "email")) {
+    kustomerPayload.emails = kustomerPayload.emails
+      ? kustomerPayload.emails
+      : [{ type: "home", email: getFieldValueFromMessage(message, "email") }];
+  }
+
+  if (getFieldValueFromMessage(message, "phone")) {
+    kustomerPayload.phones = kustomerPayload.phones
+      ? kustomerPayload.phones
+      : [{ type: "home", phone: getFieldValueFromMessage(message, "phone") }];
+  }
+  if (getFieldValueFromMessage(message, "website")) {
+    kustomerPayload.urls = [
+      {
+        url: getFieldValueFromMessage(message, "website")
+      }
+    ];
+  }
+
+  const address = getFieldValueFromMessage(message, "address");
+  if (address) {
+    let addrStr;
+    if (typeof address === "string") {
+      addrStr = address;
+    } else {
+      const { street, city, state, postalCode } = address;
+      addrStr =
+        street && city && state && postalCode
+          ? `${street}, ${city}, ${state}, ${postalCode}`
```

Comment:
Should we consider partial updates also? I mean, if only street and city is available, we send only those.

Reply:
fixed